packages/sdk/react-shell/src/panels/IdentityPanel/steps/AgentPanel.tsx

```diff
@@ -0,0 +1,161 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { useState, useCallback, useEffect } from 'react';
+import { type Event, type SingleOrArray } from 'xstate';
+
+import { useClient, useAgentHostingProviderClient } from '@dxos/react-client';
+import { type Identity, useHaloInvitations } from '@dxos/react-client/halo';
+import { Invitation, InvitationEncoder } from '@dxos/react-client/invitations';
+import { useTranslation } from '@dxos/react-ui';
+
+import { Action, Actions, StepHeading, Input } from '../../../components';
+import { type IdentityPanelStepProps } from '../IdentityPanelProps';
+import { type IdentityEvent } from '../identityMachine';
+
+export interface AgentFormProps extends Omit<IdentityPanelStepProps, 'send'> {
+  send?: (event: SingleOrArray<Event<IdentityEvent>>) => void;
+  onAgentCreate?: () => Promise<void>;
+  onAgentDestroy?: () => Promise<void>;
+  identity?: Identity;
+}
+
+export type AgentFormImplProps = AgentFormProps & {
+  agentStatus?: string;
+  agentActive?: boolean;
+  validationMessage?: string;
+};
+
+export const AgentForm = (props: AgentFormProps) => {
+  const { identity } = props;
+  // const { t } = useTranslation('os');
+  const [validationMessage, setValidationMessage] = useState('');
+  const [agentStatus, setAgentStatus] = useState('');
+  const [agentActive, setAgentActive] = useState(false);
+  const client = useClient();
+  const invitations = useHaloInvitations();
+  const agentHostingProviderClient = useAgentHostingProviderClient();
+
+  useEffect(() => {
+    const fetchAgentStatus = async () => {
+      // TODO(nf): if agent exists, subscribe to updates instead of oneshot status
+      const agentStatus = await agentHostingProviderClient?.getAgent(identity?.identityKey.truncate() ?? 'foo');
+      if (agentStatus) {
+        setAgentActive(true);
+        setAgentStatus(agentStatus);
+      } else {
+        setAgentActive(false);
+        setValidationMessage('No agent deployed via provider.');
+      }
+    };
+    void fetchAgentStatus();
+  }, []);
+
+  const onAgentCreateEvent = useCallback(async (invitation: Invitation) => {
```

Comment:
```
const handleX = () => {};
return <Foo onX={handleX} />;
```

Reply:
I'm not sure if I follow, possibly due to my poor function naming. `onAgentCreateEvent` is passed to `invitation.subscribe()` within `handleAgentCreate` which is referred to in `<AgentFormImpl... onAgentCreate={handleAgentCreate}`